From: Julian Braha <julianbraha@gmail.com>
Date: Mon, 17 Aug 2026 22:03:11 +0100
Subject: [PATCH] lib: cleanup "fake" tristates in Kconfig

These 7 DECOMPRESS_ options (e.g.  DECOMPRESS_GZIP) currently have the
tristate type, but can never be set to M.  Their only valid values are Y
and N, making them effectively booleans.

Let's make their types more accurate by changing them to 'bool'.  Note
that this is only a code cleanup, there is no functional change.

These bistates were found by kconfirm, a static analysis tool for Kconfig.

Link: https://lore.kernel.org/20260817210311.2142999-1-julianbraha@gmail.com
Signed-off-by: Julian Braha <julianbraha@gmail.com>
Cc: Arnd Bergmann <arnd@arndb.de>
Cc: Jani Nikula <jani.nikula@linux.intel.com>
Cc: Julia Lawall <julia.lawall@inria.fr>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 lib/Kconfig | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/lib/Kconfig b/lib/Kconfig
index 4e6b34c3346d5..1e42f5167c68b 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -219,29 +219,29 @@ source "lib/xz/Kconfig"
 #
 config DECOMPRESS_GZIP
 	select ZLIB_INFLATE
-	tristate
+	bool
 
 config DECOMPRESS_BZIP2
-	tristate
+	bool
 
 config DECOMPRESS_LZMA
-	tristate
+	bool
 
 config DECOMPRESS_XZ
 	select XZ_DEC
-	tristate
+	bool
 
 config DECOMPRESS_LZO
 	select LZO_DECOMPRESS
-	tristate
+	bool
 
 config DECOMPRESS_LZ4
 	select LZ4_DECOMPRESS
-	tristate
+	bool
 
 config DECOMPRESS_ZSTD
 	select ZSTD_DECOMPRESS
-	tristate
+	bool
 
 #
 # Generic allocator support is selected if needed
-- 
2.47.3

